spec/registry.md

```diff
@@ -255,3 +255,550 @@ which only expects the `accord` option:
 > .input {$object :noun case=nominative}
 > {{You see {$color :adjective article=indefinite accord=$object} {$object}!}}
 >```
+
+# Default Registry
+
+This section describes the functions which each implementation MUST provide
+to be conformant with this specification.
+
+## String Value Selection and Formatting
+
+### Functions
+
+The following functions are provided:
+
+The function `:string` provides string selection and formatting.
+
+### Operands
+
+The _operand_ is any literal or an implementation-defined set of string or
```

Comment:
```suggestion
The _operand_ is any string generated by the
`*(quoted-char / quoted-escape)`, `name`, or `number-literal` productions in the grammar;
or an implementation-defined set of string or
```

Reply:
I see what you're doing!

The `name` or `number-literal` productions are narrower than the first description (which is the guts of a quoted literal). That narrower production defines what's allowed in an unquoted literal, which has no real bearing here (it's a detail of the syntax). 

The verb "generated" also bothers me, as one doesn't generate the operand. I kind of want to say the "resolved value of a _literal_", since one of the things that should happen is that the `quoted-escape` becomes unescaped in the data-model or in-memory representation. It's only escaped in the syntax to avoid conflicting with the grammar.

Perhaps:

```suggestion
The _operand_ is any _literal_'s value less any quotes and with any `quoted-escape` sequences
converted to their escaped characters
or an implementation-defined set of string or
```

We should add an example too.

Reply:
I rewrote and added an example. Thanks!

Reply:
I was confused as to what you meant by a 'resolved value', and could see two possible interpretations. My comment was specific to :string (because as you say the results are always string comparisons anyway).

The spec always has to work in terms of strings, because it knows nothing about datatypes, and can't require particular datatypes because they vary by environment and implementation capabilities.

However, I would expect many MF2 implementations to actually operate with inputs that are not strings, but rather datatypes: both primitive and non-primitive. We can't have a spec that precludes implementations of MF2 from doing that. And those implementations will want to keep the values in datatypes where feasible, instead of always stringifying.

So, for example, it should be possible for an implementation to compile a MF2 into a data structure whereby when it knows that $input is a long, it can convert a literal for matching also into a long in that data structure, and just do long comparison.

Similarly, for plural matching, the compiled data structure could contain an enum value `few', instead of converting the an internal value for the plural category into a string and then doing a string comparison.